app/src/main/kotlin/com/wire/android/ui/home/conversations/details/options/GroupConversationOptions.kt

```diff
@@ -113,126 +123,163 @@ fun GroupConversationSettings(
-                isSwitchEnabled = state.isUpdatingReadReceiptAllowed,
-                switchState = state.isReadReceiptAllowed,
-                isLoading = state.loadingReadReceiptOption,
-                onCheckedChange = onReadReceiptSwitchClicked
+                }
             )
         }
+    }
+}
+
+/**
+ * Adds a new section to the LazyListScope with a header and items.
+ * @param folderTitleResId The resource ID for the folder title.
+ * @param showFolder Whether to show the folder header. Defaults to true.
+ * @param items A list of composable functions representing the items to be displayed in the section.
+ *              If the list is empty, the whole section will be skipped, so that there will be no header with empty section.
+ */
+private fun LazyListScope.folderWithItems(
+    @StringRes folderTitleResId: Int,
+    showFolder: Boolean = true,
+    items: List<@Composable () -> Unit> = emptyList(),
+) {
+    if (items.isNotEmpty() && showFolder) {
         item {
-            ConversationProtocolDetails(
-                protocolInfo = state.protocolInfo
-            )
-        }
-        if (state.isWireCellFeatureEnabled) {
-            item {
-                ConversationCellDetails(
-                    isWireCellEnabled = state.isWireCellEnabled,
-                    isLoading = state.loadingWireCellState,
-                    onCheckedChange = onWireCellSwitchClicked,
+            FolderHeader(
+                name = stringResource(folderTitleResId),
+                padding = PaddingValues(
+                    horizontal = dimensions().spacing16x,
+                    vertical = dimensions().spacing8x - (2 * DividerDefaults.Thickness),
                 )
-            }
+            )
         }
     }
+    items(count = items.size) { index ->
+        items[index].invoke()
+    }
 }
 
-@Composable
-fun ConversationProtocolDetails(
-    protocolInfo: Conversation.ProtocolInfo,
-    modifier: Modifier = Modifier
-) {
-    Column(modifier = modifier) {
-        FolderHeader(name = stringResource(R.string.folder_label_protocol_details))
+private fun <E> MutableList<E>.addIf(condition: Boolean, element: E) {
+    if (condition) add(element)
+}
 
+private fun conversationProtocolDetailsItems(
+    protocolInfo: Conversation.ProtocolInfo,
+): List<@Composable () -> Unit> = buildList {
+    add {
         ProtocolDetails(
             label = UIText.StringResource(R.string.protocol),
             text = UIText.DynamicString(protocolInfo.name())
         )
+    }
 
-        if (protocolInfo is Conversation.ProtocolInfo.MLS) {
+    if (protocolInfo is Conversation.ProtocolInfo.MLS) {
+        add {
             ProtocolDetails(
                 label = UIText.StringResource(R.string.cipher_suite),
                 text = UIText.DynamicString(protocolInfo.cipherSuite.toString())
             )
+        }
 
-            if (BuildConfig.PRIVATE_BUILD) {
+        if (BuildConfig.PRIVATE_BUILD) {
+            add {
```

Comment:
Question: Perhaps the custom function `addIf` can be used here to standardize, at least for this file 🤔